### PR TITLE
Switch to new infologger macro to specify level and severity at once (QC-396)

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

File: Framework/src/CheckRunner.cxx [hunk 1]
```diff
@@ -139,12 +139,13 @@ CheckRunner::CheckRunner(std::vector<Check> checks, std::string configurationSou
...
     mTotalNumberMOStored(0)
 {
+  ILOG.setFacility("Check");
   try {
     mConfigFile = ConfigurationFactory::getConfiguration(configurationSource);
   } catch (...) {
     // catch the exceptions and print it (the ultimate caller might not know how to display it)
-    ILOG(Fatal) << "Unexpected exception during initialization:\n"
-                << boost::current_exception_diagnostic_information(true) << ENDM;
+    ILOG << LogFatalSupport << "Unexpected exception during initialization:\n"
```
Comment (knopers8) [inline Framework/src/CheckRunner.cxx:147]: I see that here exceptions are logged as fatal, while in Checks they are just errors (however, also fatal in fact). Is there any reason for this difference?

File: Framework/src/ServiceDiscovery.cxx [hunk 2]
```diff
@@ -124,7 +124,7 @@ void ServiceDiscovery::runHealthServer(unsigned int port)
     }
   } catch (std::exception& e) {
     mThreadRunning = false;
-    ILOG(Error) << "ServiceDiscovery::runHealthServer - " << e.what() << ENDM;
+    std::cerr << "ServiceDiscovery::runHealthServer - " << e.what() << std::endl;
```
Comment (knopers8) [inline Framework/src/ServiceDiscovery.cxx:127]: why cerr?

File: Framework/src/runPostProcessingOCC.cxx [hunk 3]
```diff
@@ -91,10 +91,10 @@ class PostProcessingOCCStateMachine : public RuntimeControlledObject
     try {
       mRunner->start();
     } catch (const std::exception& ex) {
-      ILOG(Error) << "Exception caught: " << ex.what() << ENDM;
+      ILOG << LogErrorSupport << "Exception caught: " << ex.what() << ENDM;
```
Comment (knopers8) [inline Framework/src/runPostProcessingOCC.cxx:94]: Why some catch statements have the "support" severity, while others have "Ops"?

File: Modules/MFT/src/BasicTrackQcTask.cxx [hunk 4]
```diff
@@ -37,11 +37,11 @@ BasicTrackQcTask::~BasicTrackQcTask()
 
 void BasicTrackQcTask::initialize(o2::framework::InitContext& /*ctx*/)
 {
-  ILOG(Info) << "initialize BasicTrackQcTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+  ILOG << LogInfoSupport << "initialize BasicTrackQcTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
 
   // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
   if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
-    ILOG(Info) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+    ILOG << LogInfoSupport << "Custom parameter - myOwnKey: " << param->second << ENDM;
```
Comment (knopers8) [inline Modules/MFT/src/BasicTrackQcTask.cxx:44]: devel

File: Modules/Skeleton/src/SkeletonTask.cxx [hunk 5]
```diff
@@ -31,11 +31,11 @@ SkeletonTask::~SkeletonTask()
 
 void SkeletonTask::initialize(o2::framework::InitContext& /*ctx*/)
 {
-  ILOG(Info) << "initialize SkeletonTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+  ILOG << LogInfoSupport << "initialize SkeletonTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
 
   // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
   if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
-    ILOG(Info) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+    ILOG << LogInfoSupport << "Custom parameter - myOwnKey: " << param->second << ENDM;
```
Comment (knopers8) [inline Modules/Skeleton/src/SkeletonTask.cxx:38]: devel

**Review by knopers8 — CHANGES_REQUESTED** [review]
Can we have something like this?
```
ILOG(Info, Ops) << "my log" << ENDM;
```
with some defaults for the severity (`support` or `ops` being the default perhaps?)

**Comment by Barthelemy** [comment]
> Can we have something like this?

We can I would probably provide it as a global function rather than a macro. There are variadic parameters for macro nowadays but the resulting code is awful.

File: Framework/src/runPostProcessingOCC.cxx (hunk 3 above)
Comment (Barthelemy) [reply Framework/src/runPostProcessingOCC.cxx:94]: Indeed, I have put all of them with Support

File: Framework/src/CheckRunner.cxx (hunk 1 above)
Comment (Barthelemy) [reply Framework/src/CheckRunner.cxx:147]: not really, I will try to review the ones that need fatal.

Comment (Barthelemy) [reply Framework/src/CheckRunner.cxx:147]: I will also send them to the Operator. If something does not start or simply stops, they should know.

File: Framework/src/ServiceDiscovery.cxx (hunk 2 above)
Comment (Barthelemy) [reply Framework/src/ServiceDiscovery.cxx:127]: no idea

**Comment by knopers8** [comment]
> This being said, the number of characters to type is roughly the same (3 characters difference, which I could easily make disappear as I can remove the prefix "Log"). Is there any other arguments in favour of one option or the other ?

1. Having default severities - even less typing.
2. More familiar interface, which resembles the one of fairlogger - less confusion.
3. Just my personal esthetic feelings ;) Having concatenated arguments like "ErrorOps" seems less neat to me than putting them as separate arguments in `ILOG(Error, Ops)`.

However, if we go that way, `Ops` or `Support` keywords shouldn't be just substituted by the preprocessor, because it is likely that someone uses them in other context.

I guess that removing the "Log" could also make it shorter and wouldn't be that bad. I let you decide, with my arguments convincing you or not.

**Comment by Barthelemy** [comment]
Ok, done, ILOG can take 1 or 2 arguments.

File: Framework/include/QualityControl/QcInfoLogger.h [hunk 6]
```diff
@@ -58,14 +62,25 @@ class QcInfoLogger : public AliceO2::InfoLogger::InfoLogger
...
-#define ILOG(severity) ILOG_INST << AliceO2::InfoLogger::InfoLogger::Severity::severity
 #define ILOGI ILOG_INST << AliceO2::InfoLogger::InfoLogger::Info
 #define ILOGW ILOG_INST << AliceO2::InfoLogger::InfoLogger::Warning
 #define ILOGE ILOG_INST << AliceO2::InfoLogger::InfoLogger::Error
 #define ILOGF ILOG_INST << AliceO2::InfoLogger::InfoLogger::Fatal
 #define ENDM AliceO2::InfoLogger::InfoLogger::endm;
 
+#define NUM_ARGS_(_1, _2, _3, _4, _5, _6, _7, _8, TOTAL, ...) TOTAL
+#define NUM_ARGS(...) NUM_ARGS_(__VA_ARGS__, 6, 5, 4, 3, 2, 1, 0)
+#define CONCATE_(X, Y) X##Y
```
Comment (knopers8) [inline Framework/include/QualityControl/QcInfoLogger.h:75]: concatenate?

**Comment by knopers8** [comment]
Do you know if i can see it working in DPL? I try with adding `--infologger-severity support` to `o2-qc-run-basic`, but all logs seem to be present anyway.

File: Framework/include/QualityControl/QcInfoLogger.h (hunk 6 above)
Comment (Barthelemy) [reply Framework/include/QualityControl/QcInfoLogger.h]: ok

**Comment by Barthelemy** [comment]
Do you mean to have the output of DPL redirected to infologger ? 
I have not tested it directly. I am not sure what is this argument `--infologger-severity`. I tested with FairLogger though.

**Comment by knopers8** [comment]
> Do you mean to have the output of DPL redirected to infologger ?

Probably the other way. To use the ILOG macros and see the logs the terminal, running DPL. `infologger-severity` should filter ot logs which are more "severe" than indicated.

**Comment by knopers8** [comment]
Anyway, this is probably a separate issue, which should be understood by Giulio and Sylvain.

**Review by knopers8 — APPROVED** [review]

**Comment by Barthelemy** [comment]
I agree that it is a different issue. I am fixing the format and then merging.

**Comment by Barthelemy** [comment]
@MichaelLettrich There seems to be a problem with the mac builders.

**Comment by MichaelLettrich** [comment]
@Barthelemy there seems to be a problem that git has on macos with the LLVM repo - cloning/ updating sporadically fails. @ktf changed the build scripts in a way that should fix the issue. I relaunched the builders after an update.

**Pull request merged**